lib/kino/table.ex

```diff
@@ -120,20 +120,16 @@ defmodule Kino.Table do
 
   def handle_event("filter_by", %{"key" => key} = filter, ctx) do
     filters = if ctx.assigns.filters, do: ctx.assigns.filters, else: []
-    idx = Enum.find_index(filters, &(&1["key"] == key))
-    updated_filters = if idx, do: List.replace_at(filters, idx, filter), else: filters ++ [filter]
-    keys = Enum.map(filters, &lookup_key(ctx, &1["key"])) |> Enum.all?()
+    updated_filters = Enum.reject(filters, &(&1["key"] == key)) |> Kernel.++([filter])
+    keys = Enum.map(updated_filters, &lookup_key(ctx, &1["key"])) |> Enum.all?()
```

Comment:
We only need to check the key received in this event, the other ones should be fine.

FWIW I can't really think of a scenario where we would receive an invalid stringified key, but I see that's what I did for order_by initially, probably just in case 🙈

Reply:
Yap and talking about lookup, in the filters I end up using it only for validation because I already send the column title. But we could follow the same approach as `order_by ` passing only the key and mapping…

Reply:
We need the original key when calling `get_data` on the `Kino.Table` implementation, no?

Reply:
Yep, but just like in `order`, we could have a map to -> map back.
So, I think the only advantage is consistency if you think it’s worth it.

Reply:
So what you mean is that we don't use `key_to_string` for filters? We want to always map columns to strings when sending them to the client, because columns can technically be any term.

Reply:
Exactly! As in Explorer we have `name` as key and `to_string(name)` as title and `filter_with` already uses the string version, I’m ignoring it. But we’ll make the api public soon! I’m just sending both and using key for comparisons and title for the filter itself. I guess I should change it 🙈

Reply:
`Kino.Table` is Explorer-agnostic, so we should not make any assumptions :)

Reply:
Yep!